core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -259,7 +259,7 @@ oppia.factory('TopicEditorStateService', [
         var index = _getSubtopicPageIndex(subtopicPageId);
         var newIndex = _getNewSubtopicPageIndex(subtopicPageId);
         if (index === null) {
-          throw Error('The given subtopic page doesn\'t exist');
+          return;
```

Comment:
Why this change? In general, methods should never return undefined. But also, why not throw an error?

Reply:
Earlier, a subtopic page would have been loaded when a subtopic was created, but now, since subtopic page edit is in the modal, it is only fetched from the backend when the user wants to edit a subtopic page. So, if a subtopic is deleted without its page being fetched, an error need not be thrown.
Also, the error that subtopic doesn't exist would be thrown by deleteSubtopic itself before this is called.

Reply:
But then would there ever be a case with "index === null"?

Reply:
Yeah, if subtopics are deleted without loading their subtopic pages (which is possible now), the _getSubtopicPageIndex() would return null. It does not mean that the subtopic page doesn't exist, just that the subtopic page was never loaded from the backend.

Reply:
newSubtopicPages is only a subset of subtopicPages, so that wouldn't happen. Will add a comment explaining this.

Reply:
Oh, so not all subtopicPages are "loaded", then. Maybe _cachedSubtopicPages would be a better name.

Alternatively is it worth having internal variables that are _loadedSubtopicPages and _newSubtopicPages (and then when you need to do something that involves all subtopic pages, concatenating those lists)?

Reply:
I think all commands related to subtopic pages as a whole would require both of them. since there is no difference to the controllers between them. The only use of it is to know whether a deleted subtopic page is a new one or not to handle the subtopic ids of the following subtopic pages.
Actually, for this purpose, I think the index of the new subtopic pages in the _cachedSubtopicPages array would be enough instead of storing a duplicate of the entire subtopic page. Shall I do that?

Reply:
Yep, I'm fine with that. Let's rename to _cachedSubtopicPageIds if we're doing that, though.

Reply:
But the main subtopic pages array would still be storing the subtopic page objects, only the newSubtopicPages would be changed to just store the indices of the new subtopic pages in the main array. So, shouldn't it be _cachedSubtopicPages itself for the main array?

Reply:
Sorry, I think I misunderstood you!

Hm, I think it's better to store IDs instead of indexes in general (and you might want to use a dict data structure if that's easier). Indexes don't really have an intrinsic meaning...

Reply:
Ok, will store IDs of the new subtopic pages.

Reply:
Done.